File: java/Euler6.java
/*
Find the difference between the sum of the squares of the first one hundred natural numbers 
and the square of the sum.*/
public class Euler6 {

	
	public static void main(String[] args) {
		
		System.out.println(solution(100));
		
		
	}
	
	
	public static long solution(long num){
		return squareOfSumOfN(num) - sumOfSquareOfN(num); 
	}
	public static long sumOfSquareOfN(long n){
		return ((n * (n+1) * ((2*n)+1)) / 6);
	}

	public static long squareOfSumOfN(long n){
		return square(((n*(n+1)) / 2));
	}
	
	public static long square(long n){
		return n * n;
	}
}

File: ruby/Euler6.rb
# Find the difference between the sum of the squares of the first one hundred natural numbers 
# and the square of the sum.


class Euler6
  
  @square = Proc.new {|n| n*n}
  @squareOfSumOfN = Proc.new {|n| @square.call((n*(n+1))/2)}
  @sumOfSquareOfN = Proc.new {|n| ((n * (n+1) * ((2*n)+1)) / 6)}  

  def self.solution(num)
    @squareOfSumOfN.call(num) - @sumOfSquareOfN.call(num)
  end

end


puts Euler6.solution(100)
